private_sharing/api_views.py

```diff
@@ -254,6 +264,9 @@ def save_datatypes(self, data_file):
 
         datatypes can be looked up either via name or ID
         """
+        if self.project.auto_add_datatypes:
```

Comment:
I think we should do this only if datatypes weren't specified.

Reply:
Roger